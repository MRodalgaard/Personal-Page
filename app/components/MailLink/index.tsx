import mailImage from 'assets/img/mail.png';
import styled from 'styled-components';
import { AnalyticsEvent, logEvent } from 'utils/analytics';

const Link = styled.a`
  background: url(${mailImage.src});
  background-size: 64px 64px;
  width: 64px;
  height: 64px;
  filter: ${({ color, theme }) => (color === theme.primary ? 'grayscale(1)' : 'none')};

  &:hover {
    filter: none;
  }
`;

export default function MailLink({ color, href }: { color?: string; href: string }) {
  const onClick = () => {
    logEvent(AnalyticsEvent.MailClick);
  };

  return <Link color={color} href={href} onClick={onClick} aria-label="Mail" />;
}
