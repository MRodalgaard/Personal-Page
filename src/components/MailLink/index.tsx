import * as React from "react";
import styled from "styled-components";
import mailGreyImg from "../../assets/img/mail-grey.png";
import mailImg from "../../assets/img/mail.png";
import { AppColors } from "../../util/theme";

const Link = styled.a`
  background-image: url(${props =>
    props.color === AppColors.primary ? mailGreyImg : mailImg});
  background-size: 64px 64px;
  width: 64px;
  height: 64px;
  float: right;

  &:hover {
    background-image: url(${mailImg});
  }
`;

interface IProps {
  color?: AppColors;
  href: string;
}

const MailLink = (props: IProps) => {
  const { color, href } = props;

  return <Link color={color} href={href} />;
};

export default MailLink;
