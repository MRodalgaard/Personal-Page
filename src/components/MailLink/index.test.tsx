import { shallow } from 'enzyme';
import * as React from 'react';
import MailLink from '.';

describe('MailLink', () => {
  it('renders', () => {
    const wrapper = shallow(<MailLink href="#" />);
    expect(wrapper).toMatchSnapshot();
  });
});
