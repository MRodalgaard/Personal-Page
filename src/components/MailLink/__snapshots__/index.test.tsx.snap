// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailLink renders 1`] = `
<styled.a
  href="#"
  onClick={[Function]}
/>
`;
